compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -5361,6 +5361,22 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
         case UNWRAP_FROM_THROWABLE =>
           // js.special.unwrapFromThrowable(arg)
           js.UnwrapFromThrowable(genArgs1)
+
+        case LINKTIME_PROPERTY =>
+          // js.special.linkTimeProperty[T]("...")
+          val arg = genArgs1
+          arg match {
+            // tree.tpe will be type erasured :(
```

Comment:
There are ways to deal with that. See for example what we do for `js.constructorOf[T]`.

But before going there, consider the suggestion from https://github.com/scala-js/scala-js/pull/5000#issuecomment-2308901301 not expose the generic `linkTimeProperty` for now, and instead hard-coding the body of the few relevant methods from `scala.scalajs.LinkingInfo`.

Also, even for the generic version, perhaps it's better to have separate methods for the ~3 types we expect to allow: `Int`, `Boolean` and perhaps one day `String`.

Reply:
Right, defined three different methods for the types `linkTimeProperty(Int|Boolean|String)`